tests/test_shapes.py

```diff
@@ -105,62 +106,89 @@ def test_shapes():
     )
 
 
-def test_tpms():
-    elem = microgen.shape.newGeometry(
-        shape="tpms",
-        param_geom={
-            "surface_function": microgen.shape.surface_functions.gyroid,
-            "offset": 0.3,
-            "cell_size": (1, 2, 1),
-            "repeat_cell": (1, 2, 1),
-            "resolution": 10,
-        },
-    )
-    _ = elem.surface
-    _ = elem.surface
-    _ = elem.skeletals
-    sheet = elem.sheet
-    _ = elem.sheet
-    assert 0 < sheet.extract_surface().volume < np.prod(elem.cell_size) * np.prod(elem.repeat_cell)
-    _ = elem.generateVtk(type_part="lower skeletal")
-    _ = elem.generateVtk(type_part="upper skeletal")
-    _ = elem.generateVtk(type_part="surface")
-    _ = elem.generate(type_part="sheet")
+@pytest.mark.parametrize("type_part", ["lower skeletal", "upper skeletal", "sheet"])
+def test_tpms_given_cadquery_vtk_shapes_volume_must_be_equivalent(type_part: str):
+    # Arrange
+    tpms = microgen.Tpms(
+        surface_function=microgen.surface_functions.gyroid,
+        offset=1.0,
+    )
 
-    def graded_density(x, y, z):
-        return x
-    elem = microgen.shape.tpms.CylindricalTpms(
-        radius=1.0,
-        surface_function=microgen.shape.surface_functions.schwarzD,
-        offset=graded_density,
-        phase_shift=(0.1, 0.2, 0.3),
-        cell_size=(1, 2, 1),
-        repeat_cell=(2, 1, 1),
-        resolution=20,
+    # Act
+    shape = tpms.generate(type_part=type_part, smoothing=0, verbose=False)
+    shape_check = tpms.generateVtk(type_part=type_part)

```

Comment:
Maybe try name your shapes variables `shape_pyvista` (or `shape_vtk`) and `shape_cadquery` to be more explicit.